Actions to automate some contribution processes

risvh opened the pull request:
readable-data-changes.yml

Triggered on pull requests. Parses object, transition and category changes, substituting object IDs with object names. And posts the results as a comment under the pull request.

renumber-objects.yml

Manually triggered. Renumbers objects with ID larger than nextObjectNumber. And then sets nextObjectNumber.txt to the new max ID.

risvh committed:
Python script and github action yaml to summarize OneLifeData7 pull requests in more readable format

connorhsm committed:
Mostly semantic updates

risvh committed:
Do git diff within python to get rid of changed-files action

risvh committed:
Action to renumber objects

The action will renumber objects with ID greater than nextObjectNumber, and set nextObjectNumber.txt to the max ID post-renumbering.

Note that the original node script from CCM has the following arguments:
RenumberObjects.js <start_number> <offset?> <end_number?>

This action determines start_number automatically as the smallest ID larger than nextObjectNumber.
Run RenumberObjects.js nextObjectNumber.
(At this point the script changes nextObjectNumber.txt back to start_number. It does this because of CCM's workflow.)
That's why the action then set nextObjectNumber.txt to max ID.

connorhsm commented on .github/workflows/renumber-objects.yml:
This needs a better trigger and potentially a custom flow. 

Ideally, renumber doesn't happen after merge because thats a change after we review it and it might not renumber how we expect. We can brainstorm this some more or see if this is good enough for now.

connorhsm commented:
@risvh and fix for your failing check https://github.com/thollander/actions-comment-pull-request/issues/181#issuecomment-1407689889

risvh committed:
Made commenting on PR work when the PR's creator doesn't have write access to our repo

Relevant read:

Resource not accessible by integration
https://github.com/actions/first-interaction/issues/10#issuecomment-545576314
https://github.com/actions/first-interaction/issues/10#issuecomment-670968624

Keeping your GitHub Actions and workflows secure Part 1: Preventing pwn requests
https://securitylab.github.com/research/github-actions-preventing-pwn-requests/

risvh committed:
Fixed typo in RenumberSprites.js

risvh committed:
Renumber sprites as well
